### PR TITLE
Create KeyManagerAppConfig

### LINKED ISSUE #2255: Create KeyManagerAppConfig
https://github.com/bitcoin-s/bitcoin-s/pull/2252#discussion_r520660092

This allows us to consolidate key manager configuration and then use the `KeyManagerAppConfig` inside things like `DLCOracleAppConfig` and `WalletAppConfig`

### PR DESCRIPTION
**Pull request by benthecarman**
Closes #2255

### CODE REVIEW

File: key-manager/src/main/scala/org/bitcoins/keymanager/config/KeyManagerAppConfig.scala [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  override def configOverrides: List[Config] = confs.toList
+
+  override type ConfigType = KeyManagerAppConfig
+
+  override def newConfigOfType(
+      configOverrides: Seq[Config]): KeyManagerAppConfig =
+    KeyManagerAppConfig(directory, configOverrides: _*)
+
+  override def moduleName: String = "key-manager"
```
Comment (Christewart) [inline key-manager/src/main/scala/org/bitcoins/keymanager/config/KeyManagerAppConfig.scala]: I believe we use this for database names? I think the `-` might cause us issues in the future with various things that don't allow `-` in their namespace. I would recommend just doing `keymanager` for simplicity.

**Review by Christewart — COMMENTED** [review]
I think everything else looks ok, but the documentation needs to be updated.

File: key-manager/src/main/scala/org/bitcoins/keymanager/config/KeyManagerAppConfig.scala [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+  /** The path to our encrypted mnemonic seed */
+  lazy val seedPath: Path = {
+    baseDatadir.resolve(WalletStorage.ENCRYPTED_SEED_FILE_NAME)
+  }
+
+  override def start(): Future[Unit] = FutureUtil.unit
+
+  lazy val aesPasswordOpt: Option[AesPassword] = {
+    val passOpt = config.getStringOrNone("bitcoin-s.key-manager.aesPassword")
```
Comment (Christewart) [inline key-manager/src/main/scala/org/bitcoins/keymanager/config/KeyManagerAppConfig.scala]: Shouldn't these be incorrect now? Also a little concerning that unit tests are failing from this...

Comment (benthecarman) [reply key-manager/src/main/scala/org/bitcoins/keymanager/config/KeyManagerAppConfig.scala]: The `moduleName` is only used for db stuff so it doesn't effect anything for the key manager app config

Comment (Christewart) [reply key-manager/src/main/scala/org/bitcoins/keymanager/config/KeyManagerAppConfig.scala]: ok but why not be consistent? now I need to remember configuration is `key-manager` and database connections need to be done with `keymanager`

Comment (Christewart) [reply key-manager/src/main/scala/org/bitcoins/keymanager/config/KeyManagerAppConfig.scala]: We even use `moduleName` for database configuration it seems. While there isn't currently a database component to the key manager, there may be times in the future where we would want one 

https://github.com/bitcoin-s/bitcoin-s/blob/2de17bb4e47961fd3fe4e55d75909b42470e3c59/db-commons/src/main/scala/org/bitcoins/db/AppConfig.scala#L165-L164

**Review by Christewart — APPROVED** [review]

**Pull request merged**